modules/i3s/src/lib/parsers/parse-i3s-tile-content.js

```diff
@@ -445,3 +452,78 @@ function flattenFeatureIdsByFaceRanges(normalizedFeatureAttributes) {
 
   normalizedFeatureAttributes.id.value = orderedFeatureIndices;
 }
+
+/**
+ * Flatten feature ids using featureIndices
+ * @param {object} attributes
+ * @param {object} tile
+ * @returns {void}
+ */
+function flattenFeatureIdsByFeatureIndices(attributes, tile) {
+  const loadedFeatureIds = tile.userData.layerFeaturesAttributes.find(
+    attributesObject => attributesObject.OBJECTID
+  );
+
+  if (loadedFeatureIds && loadedFeatureIds.OBJECTID) {
+    attributes.id.value = generateFeatureIdsUsingFeatureIndices(
+      loadedFeatureIds.OBJECTID,
+      attributes.id.value
+    );
+  }
+}
+
+/**
+ * Generate feature ids using featureIndices
+ * @param {Float32Array} loadedFeatureIds
+ * @param {Float32Array} featureIndices
+ * @returns {Float32Array}
+ */
+function generateFeatureIdsUsingFeatureIndices(loadedFeatureIds, featureIndices) {
+  const featureIds = new Float32Array(featureIndices.length);
+
+  for (let index = 0; index < featureIndices.length; index++) {
+    featureIds[index] = loadedFeatureIds[featureIndices[index]];
+  }
+  return featureIds;
+}
+
+/**
+ * Load layer feature attributes and put them to user Data object
+ * @param {Object} tile
+ * @param {Object} tileset
+ * @returns {Promise}
+ */
+async function loadFeatureAttributes(tile, tileset) {
+  const attributeStorageInfo = tileset.attributeStorageInfo;
+  const attributeUrls = tile.attributeUrls;
+  const attributes = [];
+
+  for (let index = 0; index < attributeStorageInfo.length; index++) {
+    const url = attributeUrls[index];
+    const attributeName = attributeStorageInfo[index].name;
+    const attributeType = getAttributeValueType(attributeStorageInfo[index]);
+
+    try {
+      const attribute = await load(url, I3SAttributeLoader, {attributeName, attributeType});
+      attributes.push(attribute);
+    } catch (error) {
+      // do nothing
+    }
+  }
```

Comment:
I recommend parallel loading to speed things up:

```suggestion
  console.time('load attributes'); // debug
  const attributeLoadPromises = [];
  for (let index = 0; index < attributeStorageInfo.length; index++) {
    const url = attributeUrls[index];
    const attributeName = attributeStorageInfo[index].name;
    const attributeType = getAttributeValueType(attributeStorageInfo[index]);
    const promise = load(url, I3SAttributeLoader, {attributeName, attributeType}));
    attributeLoadPromises.push(promise);
  }
   try {
      attributes = await Promise.all(attributeLoadPromises);
    } catch (error) {
      // do nothing
    } finally {
      console.timeEnd('load attributes'); // debug
    }
  }
```